### DISCUSSION TITLE
create smaller image version, if uploaded to assets

### DISCUSSION THREAD

**Discussion by eliasu** [statamic/cms#5846] [Q&A] [answered]
Hi all, 

I have the following problem over and over again with some clients. They would (nonetheless I told them) upload images to their sites like: 7000x5000px and 16Mb size. Great.

So I was thinking of a three step solution for this:

1) And this is mainly my question: I want to create a smaller version of the image, at the point it gets uploaded. Like, if an image is over 2000px on the longer side, generate a smaller version of it, with a max of 2000px. How could I implement this? And how to use them with Antlers?

2) I use the Add-On [Image Optimizer](https://statamic.com/addons/4rn0/imageoptimizer-v3) to run some compression tools on the generated images from point one.

3) Create a `<Picture>` Element with various image versions in different sizes, and serve the correct image to the user. Concerning how this is can be done: I opened a thread [here](https://github.com/statamic/cms/discussions/5838). 

So my main question is Point 1, and secondly I would love to know if you have better ideas, or techniques or experiences to archive this goal? How are you dealing with enormous file dimension and sizes?

Thanks!

**Comment by jasonvarga** [comment, marked as answer]
You can define presets which will be created whenever an asset is uploaded.

See https://statamic.dev/tags/glide#presets

**Reply by eliasu** [reply]
Thanks for the reply @jasonvarga 
I didn't know that this will be generated on the uploading process as well. Cool.
